src/app/clusters/commodity-tariff-server/commodity-tariff-server.h

```diff
@@ -0,0 +1,124 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/InteractionModelEngine.h>
+#include <app/MessageDef/StatusIB.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/CHIPError.h>
+
+#include <list>
+#include <string>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace CommodityTariff {
+
+typedef DataModel::Nullable<uint32_t> epoch_s;
+
+class Delegate
+{
+public:
+    Delegate()          = default;
+    virtual ~Delegate() = default;
+
+    void SetEndpointId(EndpointId aEndpoint) { mEndpointId = aEndpoint; }
+
+
+    virtual Protocols::InteractionModel::Status GetDayEntryById(DataModel::Nullable<uint32_t> aDayEntryId,
+                                                                Structs::DayEntryStruct::Type & aDayEntry) = 0;
+
+    virtual Protocols::InteractionModel::Status GetTariffComponentInfoById(DataModel::Nullable<uint32_t>  aTariffComponentId,
+                                                                           DataModel::Nullable<chip::CharSpan> & label,
+                                                                           DataModel::List<const uint32_t> & dayEntryIDs,
```

Comment:
How does the delegate know when it's safe to deallocate the buffers this stuff is pointing at?